### ISSUE TITLE
[BUILD APP PROCESS] Ubuntu IDE - Could not find dependency: libsteam_api.so

### ISSUE THREAD

**Issue by Pikku-a** [YoYoGames/GMEXT-Steamworks#30]
When using the steamworks extension and trying to run the game, I get this error: "ERROR: Could not find dependency: libsteam_api.so". Here is part of the output:

```
-- Deploying dependencies for existing files in AppDir -- 
Deploying dependencies for ELF file AppDir/usr/bin/steamworks.so 
ERROR: Could not find dependency: libsteam_api.so 
ERROR: Failed to deploy dependencies for existing files 


System.Exception: command 'unshare -mUprf sh -c 'mount -o bind "$1" "$2/tmp/" && PATH="$PATH:/usr/local/sbin:/usr/sbin:/sbin:/tmp/squashfs-root/usr/bin/" NO_STRIP=1 chroot "$2" sh -c "cd /tmp/ && linuxdeploy --appdir=AppDir/"'  -- /home/myusername/GameMakerStudio2/yyc/strategy_game_but_with_penguin /opt/steam-runtime' failed with exit status 1
   at Igor.LinuxBuilder.plink_async(String command, Boolean fail_on_error)
   at Igor.LinuxBuilder.BuildAppImage(String work_dir, String exe_path, String assets_zip_path, String projectName, Boolean package)
   at Igor.LinuxBuilder.Run()
Igor complete.
```
Another project that uses an older version of the extension (1.1.5) runs, but the steam stuff doesn't seem to work.
IDE version: 2023.800.0.404
Runtime version: 2023.800.0.425
I have downloaded and configured the steamworks sdk too.

**Comment by DiasFranciscoA** [comment]
We would need some extra information what IDE are you using macOS or Windows?

**Comment by Pikku-a** [comment]
I'm using the Ubuntu IDE.

**Comment by nkrapivin** [comment]
Let me answer:

So basically you're trying to build an AppImage for a Linux Steam game, which is something you should not do.
When creating an executable in the file extension listbox select "Zip Archive" instead of an "AppImage",
AppImage is only needed for out-of-steam distributions and it bundles the libraries into the app, Steam already has those libraries and you don't need to use AppImage.

**Comment by DiasFranciscoA** [comment]
the problem seems to be that running from the IDE also has this bug... and the log indicates that this is NOT an extension bug but a bug in the build process.

@nkrapivin is right though when building for deploying to steam you should **NOT** be be using AppImage.

But for running from IDE will build a AppImage for you, so there is something fishy going on with the build process.
We will look into the problem

**Comment by Pikku-a** [comment]
I didn't try to build it specifically as an appimage, I just tried to run the game.

When building the game (as zip) and then trying to run it, there is this error:
`error while loading shared libraries: libcrypto.so.1.0.0: cannot open shared object file: No such file or directory`

**Comment by nkrapivin** [comment]
The libcrypto error is expected because it's a part of the Steam Linux Runtime, you need to run your game through Steam (as a non steam app will work too)

**Comment by DiasFranciscoA** [comment]
fixed in 1.6.0 github release
